### DISCUSSION TITLE
File in private Github repository as .json/.yaml/html custom datasource?

### DISCUSSION THREAD

**Discussion by evilensky** [renovatebot/renovate#27707] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

How might Renovate be configured to use a private Github repository as a `data` source?  We'd like to host a json or yaml formatted file that is populated as part of a Gitops workflow, as a datasource for a custom manager.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
This could probably be done by using a custom datasource with URL like api.github.com (if private and credentials required) or raw.githubusercontent.com (if public)

**Reply by evilensky** [reply]
Thank you. It sounds like interacting with api.github.com would require using their repository `content` api and working with a transformTemplate to base64 decode and process the file.

Would you consider a request for a build in data source (`github-file` ?) to abstract away some of this?

**Reply by rarkins** [reply]
I would prefer to incorporate it into custom datasources if possible
